From a82fbcd3332cfebc41b1a514b0b1704ef57bc967 Mon Sep 17 00:00:00 2001
From: ywan171 <yi.a.wang@intel.com>
Date: Mon, 30 Jul 2012 22:22:24 +0800
Subject: Fix rtsp streaming continuously seeking, video freeze issue.

BZ: 49305

After seeking, the video rtcp packet receives too late, and the rtp packets
which receives before the rtcp packets are cached, if we don't clear these packets,
when finish the nextseeking it will use the cache rtp packets to calculate the
timestamp and it will get a wrong timestamp for A/V sync.

orig-patch:68627:(BZ:35189)Fix rtsp streaming continuously seeking, video freeze issue.

Category: aosp improvement
Domain: Video.Media-streaming
Origin: internal
Upstream-Candidate: yes

Change-Id: I288dbdded1f86edc43709cbca71790552105a004
Orig-Change-Id: Ia46fb3c286c6ffa8cbfdc12b21090727e318b997
Signed-off-by: ywan171 <yi.a.wang@intel.com>
Signed-off-by: chuzhoux <chux.zhou@intel.com>
---
 media/libstagefright/rtsp/MyHandler.h |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/media/libstagefright/rtsp/MyHandler.h b/media/libstagefright/rtsp/MyHandler.h
index f4b5846..fd569de 100644
--- a/media/libstagefright/rtsp/MyHandler.h
+++ b/media/libstagefright/rtsp/MyHandler.h
@@ -1187,6 +1187,8 @@ struct MyHandler : public AHandler {
 
                     info->mRTPAnchor = 0;
                     info->mNTPAnchorUs = -1;
+                    // clear the cached packets
+                    info->mPackets.clear();
                 }
 
                 mAllTracksHaveTime = false;
-- 
1.7.4.1

